File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
bool isAnagram(string s, string t)
{
    if (s.length() != t.length())
        return false;
    vector<int> track1(26, 0);
    vector<int> track2(26, 0);
    for (int i = 0; i < s.length(); i++)
    {
        track1[s[i] - 'a']++;
    }
    for (int i = 0; i < s.length(); i++)
    {
        track2[t[i] - 'a']++;
    }
    for (int i = 0; i < 26; i++)
    {
        if (track1[i] != track2[i])
            return false;
    }
    return true;
}
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
                if (s.length() != t.length())
            return false;
        int[] p = new int[26];
        int[] q=new int[26];
        int n = s.length();
        for (int i = 0; i < n; i++) {
            int c1 = s.charAt(i) - 'a';
            int c2 = t.charAt(i) - 'a';
            p[c1] = p[c1] + 1;
            q[c2] = q[c2] + 1;
        }
       for (int i = 0; i < p.length; i++) {
            if (p[i]!=q[i])
                return false;
        }
        return true;
    }
}